src/Tools/dotnet-dump/Dumper.cs

```diff
@@ -83,34 +83,24 @@ public int Collect(IConsole console, int processId, string output, bool diag, Du
                 }
                 console.Out.WriteLine($"Writing {dumpTypeMessage} to {output}");
 
-                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-                {
-                    // Get the process
-                    Process process = Process.GetProcessById(processId);
+                var client = new DiagnosticsClient(processId);
 
-                    Windows.CollectDump(process, output, type);
```

Comment:
The problem with this change is that now dotnet-dump collect on Windows won't work with older runtimes (<= 2.2) and it did before. May be it isn't that important because dotnet-dump collect on xplat only works runtimes > 2.2 and your change is needed to provide the condition on Windows. 

dotnet-dump could catch that the client.WriteDump fails on the unsupported Windows runtimes, but the exception is inconsistent/difficult than xplat. I think it is something like TimeoutException on Windows and PlatformNotSupportedException on xplat.

/cc: @josalem on the client assembly exception difference.